adapters/triplelift/triplelift.go

```diff
@@ -0,0 +1,147 @@
+package triplelift
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type TripleliftAdapter struct {
+	endpoint string
+}
+
+type TripleliftInnerExt struct {
+	Format int `json:"format"`
+}
+
+type TripleliftRespExt struct {
+	Triplelift TripleliftInnerExt `json:"triplelift_pb"`
+}
+
+func getBidType(ext TripleliftRespExt) openrtb_ext.BidType {
+	t := ext.Triplelift.Format
+	if t == 11 {
+		return openrtb_ext.BidTypeVideo
+	}
+	return openrtb_ext.BidTypeBanner
+}
+
+func processImp(imp *openrtb.Imp) error {
+	// get the triplelift extension
+	var ext adapters.ExtImpBidder
+	var tlext openrtb_ext.ExtImpTriplelift
+	if err := json.Unmarshal(imp.Ext, &ext); err != nil {
+		return err
+	}
+	if err := json.Unmarshal(ext.Bidder, &tlext); err != nil {
+		return err
+	}
+	if imp.Banner == nil && imp.Video == nil {
+		return fmt.Errorf("neither Banner nor Video object specified")
+	}
```

Comment:
If we accept banners, are the width and height properties relevant? Other adapters check for `nil` on the `imp.Banner.W`, `imp.Banner.H`, and `imp.Banner.Format`. Would that be useful in our context?
```
 34 func processImp(imp *openrtb.Imp) error {
 35     // get the triplelift extension
 36     var ext adapters.ExtImpBidder
 37     var tlext openrtb_ext.ExtImpTriplelift
 38     if err := json.Unmarshal(imp.Ext, &ext); err != nil {
 39         return err
 40     }
 41     if err := json.Unmarshal(ext.Bidder, &tlext); err != nil {
 42         return err
 43     }
 44     if imp.Banner == nil && imp.Video == nil {
 45         return fmt.Errorf("neither Banner nor Video object specified")
 46     }
    +   if imp.Banner {
    +   	if imp.Banner.W == nil && imp.Banner.H == nil && len(imp.Banner.Format) > 0 {
    +              firstFormat := bannerCopy.Format[0]
    +              bannerCopy.W = &(firstFormat.W)
    +              bannerCopy.H = &(firstFormat.H)
    +          }
    +    }
 47     imp.TagID = tlext.InvCode
 48     // floor is optional
 49     if tlext.Floor == nil {
```
Maybe width and height are not relevant at this point, just asking.

Reply:
Good point, but this is something we'd prefer to handle downstream from this application.